com.bkmarriott.payment/src/main/java/com/bkmarriott/payment/infrastructure/persistence/adapter/PaymentQueryAdapter.java

```diff
@@ -0,0 +1,24 @@
+package com.bkmarriott.payment.infrastructure.persistence.adapter;
+
+import com.bkmarriott.payment.application.outputport.PaymentQueryOutputPort;
+import com.bkmarriott.payment.domain.Payment;
+import com.bkmarriott.payment.infrastructure.persistence.entity.PaymentEntity;
+import com.bkmarriott.payment.infrastructure.persistence.repository.PaymentRepository;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class PaymentQueryAdapter implements PaymentQueryOutputPort {
+
+  private final PaymentRepository paymentRepository;
+
+  @Override
+  public Optional<Payment> findById(Long paymentId) {
+    return paymentRepository.findById(paymentId)
+        .map(PaymentEntity::toDomain);
```

Comment:
조회 결과가 없으면 어떻게 될까요?

```suggestion
    return paymentRepository.findById(paymentId)
        .map(PaymentEntity::toDomain)
        .or(Optional::empty)
```

조회 결과가 없어도 empty가 반환 잘 되네요? 홀리몰리